### PR TITLE
create activity movie of genre

### PR DESCRIPTION
**Pull request by anhpham2403**
create activity movie of genre

### CODE REVIEW

File: app/src/main/java/com/framgia/moviedb/screen/detail/DetailViewModel.java [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package com.framgia.moviedb.screen.detail;
 
 import android.content.Context;
+import android.content.Intent;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/detail/DetailViewModel.java]: ERROR: Unused import: android.content.Intent. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/moviedb/screen/detail/DetailViewModel.java [hunk 2]
```diff
@@ -9,6 +10,7 @@
 import com.framgia.moviedb.data.model.Genre;
 import com.framgia.moviedb.data.model.Movie;
 import com.framgia.moviedb.screen.genres.GenreAdapter;
+import com.framgia.moviedb.screen.moviesofgenre.MoviesofgenreActivity;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/detail/DetailViewModel.java]: ERROR: Unused import: com.framgia.moviedb.screen.moviesofgenre.MoviesofgenreActivity. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/moviedb/screen/genres/GenresViewModel.java [hunk 3]
```diff
@@ -66,6 +67,6 @@ public void onDestroy() {
 
     @Override
     public void onItemClickListener(Genre genre) {
-        // TODO: 28/09/2017
+        mContext.startActivity(MoviesofgenreActivity.getIntentMoviesOfGenre(mContext,genre));
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/genres/GenresViewModel.java]: ERROR: ',' is not followed by whitespace.

File: app/src/main/java/com/framgia/moviedb/screen/detail/DetailViewModel.java (hunk 1 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/detail/DetailViewModel.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.content.Intent'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/moviedb/screen/detail/DetailViewModel.java (hunk 2 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/detail/DetailViewModel.java]: WARNING: [UnusedImports] Avoid unused imports such as 'com.framgia.moviedb.screen.moviesofgenre.MoviesofgenreActivity'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/moviedb/screen/moviesofgenre/MoviesofgenreViewModel.java [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+import java.util.List;
+
+/**
+ * Exposes the data to be used in the Moviesofgenre screen.
+ */
+
+public class MoviesofgenreViewModel extends BaseObservable
+        implements MoviesofgenreContract.ViewModel, MovieAdapter.OnItemClickListener {
+    private MoviesofgenreContract.Presenter mPresenter;
+    private boolean isLoadingMore;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/moviesofgenre/MoviesofgenreViewModel.java]: WARNING: [AvoidFieldNameMatchingMethodName] Field isLoadingMore has the same name as a method
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/naming.html#AvoidFieldNameMatchingMethodName

File: app/src/main/java/com/framgia/moviedb/screen/moviesofgenre/MoviesofgenreViewModel.java [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+    private Genre mGenre;
+    private Context mContext;
+    private List<Movie> mMovies;
+    private MovieAdapter mAdapter;
+    private int mPage;
+    private RecyclerView.OnScrollListener mScrollListener = new RecyclerView.OnScrollListener() {
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/moviesofgenre/MoviesofgenreViewModel.java]: ERROR: [PMD-CPD] 20 lines duplicated.
app/src/main/java/com/framgia/moviedb/screen/movies/MoviesViewModel.java

File: app/src/main/java/com/framgia/moviedb/screen/movies/MoviesContract.java [hunk 6]
```diff
@@ -22,6 +22,6 @@
      * Presenter.
      */
     interface Presenter extends BasePresenter {
-        void getDataMovies(int category, int page);
+        void getDataMovies(int category, int page,int mIdGenre);
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/movies/MoviesContract.java]: ERROR: ',' is not followed by whitespace.

File: app/src/main/java/com/framgia/moviedb/screen/movies/MoviesFragment.java [hunk 7]
```diff
@@ -31,13 +32,23 @@ public static MoviesFragment newInstance(int category) {
...
+    }
+
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         mMovieReposity =
                 new MovieReposity(new MovieRemoteDataSource(MovieServiceClient.getInstance()));
         category = getArguments().getInt(Constant.MOVIES_BUNDLE);
-        mViewModel = new MoviesViewModel(getContext(), category);
+        idGenre = getArguments().getInt(Constant.ID_BUNDLE,-1);
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/movies/MoviesFragment.java]: ERROR: ',' is not followed by whitespace.

File: app/src/main/java/com/framgia/moviedb/screen/movies/MoviesFragment.java [hunk 8]
```diff
@@ -31,13 +32,23 @@ public static MoviesFragment newInstance(int category) {
...
+        mViewModel = new MoviesViewModel(getContext(), category,idGenre);
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/movies/MoviesFragment.java]: ERROR: ',' is not followed by whitespace.

File: app/src/main/java/com/framgia/moviedb/screen/movies/MoviesPresenter.java [hunk 9]
```diff
@@ -35,7 +35,7 @@ public void onStop() {
     }
 
     @Override
-    public void getDataMovies(int category, int page) {
+    public void getDataMovies(int category, int page,int mIdGenre) {
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/movies/MoviesPresenter.java]: ERROR: ',' is not followed by whitespace.

File: app/src/main/java/com/framgia/moviedb/screen/movies/MoviesPresenter.java [hunk 10]
```diff
@@ -52,6 +52,8 @@ public void getDataMovies(int category, int page) {
                 break;
             case Constant.FAVORITE_FRAGMENT:
                 break;
+            case Constant.MOVIE_OF_GENRES_FRAGMENT:
+                observable = mMovieReposity.getMoviesByIdGenre(mIdGenre,BuildConfig.API_KEY,page);
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/movies/MoviesPresenter.java]: ERROR: ',' is not followed by whitespace.

Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/movies/MoviesPresenter.java]: ERROR: ',' is not followed by whitespace.

File: app/src/main/java/com/framgia/moviedb/screen/movies/MoviesViewModel.java [hunk 11]
```diff
@@ -67,7 +69,7 @@ public void onStop() {
     @Override
     public void setPresenter(MoviesContract.Presenter presenter) {
         mPresenter = presenter;
-        mPresenter.getDataMovies(mCategory, mPage);
+        mPresenter.getDataMovies(mCategory, mPage,mIdGenre);
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/movies/MoviesViewModel.java]: ERROR: ',' is not followed by whitespace.

File: app/src/main/java/com/framgia/moviedb/screen/movies/MoviesViewModel.java [hunk 12]
```diff
@@ -113,7 +116,7 @@ public void setLoadingMore(boolean loadingMore) {
 
     public void loadMore() {
         mPage++;
-        mPresenter.getDataMovies(mCategory, mPage);
+        mPresenter.getDataMovies(mCategory, mPage,mIdGenre);
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/movies/MoviesViewModel.java]: ERROR: ',' is not followed by whitespace.

File: app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenreActivity.java [hunk 13]
```diff
@@ -0,0 +1,56 @@
+package com.framgia.moviedb.screen.moviesactivity;
+
+import android.content.Context;
+import android.content.Intent;
+import android.databinding.DataBindingUtil;
+import android.os.Bundle;
+import android.support.v4.app.FragmentManager;
+import com.framgia.moviedb.R;
+import com.framgia.moviedb.data.model.Genre;
+import com.framgia.moviedb.data.source.MovieReposity;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenreActivity.java]: ERROR: Unused import: com.framgia.moviedb.data.source.MovieReposity. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenreActivity.java [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+import com.framgia.moviedb.data.source.remote.MovieRemoteDataSource;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenreActivity.java]: ERROR: Unused import: com.framgia.moviedb.data.source.remote.MovieRemoteDataSource. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenreActivity.java [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+import com.framgia.moviedb.data.source.remote.service.MovieServiceClient;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenreActivity.java]: ERROR: Unused import: com.framgia.moviedb.data.source.remote.service.MovieServiceClient. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenreActivity.java [hunk 16]
```diff
@@ -0,0 +1,56 @@
...
+        return intent;
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        Genre genre = getIntent().getExtras().getParcelable(Constant.GENRES_BUNDLE);
+        setTitle(genre.getName());
+        FragmentManager fragmentManager = getSupportFragmentManager();
+        mViewModel = new MoviesofgenreViewModel(getApplicationContext(),fragmentManager, genre);
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenreActivity.java]: ERROR: ',' is not followed by whitespace.

File: app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenreContract.java [hunk 17]
```diff
@@ -0,0 +1,23 @@
+package com.framgia.moviedb.screen.moviesactivity;
+
+import com.framgia.moviedb.data.model.Movie;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenreContract.java]: ERROR: Unused import: com.framgia.moviedb.data.model.Movie. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenreContract.java [hunk 18]
```diff
@@ -0,0 +1,23 @@
...
+import com.framgia.moviedb.screen.BasePresenter;
+import com.framgia.moviedb.screen.BaseViewModel;
+import java.util.List;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenreContract.java]: ERROR: Unused import: java.util.List. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenrePresenter.java [hunk 19]
```diff
@@ -0,0 +1,31 @@
+package com.framgia.moviedb.screen.moviesactivity;
+
+import com.framgia.moviedb.BuildConfig;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenrePresenter.java]: ERROR: Unused import: com.framgia.moviedb.BuildConfig. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenrePresenter.java [hunk 20]
```diff
@@ -0,0 +1,31 @@
...
+import com.framgia.moviedb.data.model.Movie;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenrePresenter.java]: ERROR: Unused import: com.framgia.moviedb.data.model.Movie. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenrePresenter.java [hunk 21]
```diff
@@ -0,0 +1,31 @@
...
+import com.framgia.moviedb.data.source.MovieReposity;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenrePresenter.java]: ERROR: Unused import: com.framgia.moviedb.data.source.MovieReposity. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenrePresenter.java [hunk 22]
```diff
@@ -0,0 +1,31 @@
...
+import io.reactivex.android.schedulers.AndroidSchedulers;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenrePresenter.java]: ERROR: Unused import: io.reactivex.android.schedulers.AndroidSchedulers. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenrePresenter.java [hunk 23]
```diff
@@ -0,0 +1,31 @@
...
+import io.reactivex.disposables.CompositeDisposable;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenrePresenter.java]: ERROR: Unused import: io.reactivex.disposables.CompositeDisposable. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenrePresenter.java [hunk 24]
```diff
@@ -0,0 +1,31 @@
...
+import io.reactivex.observers.DisposableObserver;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenrePresenter.java]: ERROR: Unused import: io.reactivex.observers.DisposableObserver. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenrePresenter.java [hunk 25]
```diff
@@ -0,0 +1,31 @@
...
+import io.reactivex.schedulers.Schedulers;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenrePresenter.java]: ERROR: Unused import: io.reactivex.schedulers.Schedulers. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenrePresenter.java [hunk 26]
```diff
@@ -0,0 +1,31 @@
...
+import java.util.List;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenrePresenter.java]: ERROR: Unused import: java.util.List. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenreViewModel.java [hunk 27]
```diff
@@ -0,0 +1,70 @@
...
+    private Genre mGenre;
+    private Context mContext;
+    private FragmentManager mManager;
+    private Fragment mFragment;
+
+    public MoviesofgenreViewModel(Context context, FragmentManager manager, Genre genre) {
+        mGenre = genre;
+        mManager = manager;
+        mContext = context;
+        mFragment = MoviesFragment.newInstance(Constant.MOVIE_OF_GENRES_FRAGMENT,genre.getId());
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenreViewModel.java]: ERROR: ',' is not followed by whitespace.

File: app/src/main/java/com/framgia/moviedb/utils/binding/BindingUtils.java [hunk 28]
```diff
@@ -2,10 +2,16 @@
...
 import android.databinding.BindingAdapter;
 import android.support.design.widget.TabLayout;
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentManager;
+import android.support.v4.app.FragmentTransaction;
 import android.support.v4.view.ViewPager;
 import android.support.v7.widget.RecyclerView;
+import android.widget.FrameLayout;
 import android.widget.ImageView;
+import android.widget.LinearLayout;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/utils/binding/BindingUtils.java]: ERROR: Unused import: android.widget.LinearLayout. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/moviedb/utils/binding/BindingUtils.java [hunk 29]
```diff
@@ -2,10 +2,16 @@
...
 import com.framgia.moviedb.screen.main.ViewPagerAdapter;
+import com.framgia.moviedb.screen.movies.MoviesFragment;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/utils/binding/BindingUtils.java]: ERROR: Unused import: com.framgia.moviedb.screen.movies.MoviesFragment. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenreActivity.java (hunk 13 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenreActivity.java]: WARNING: [UnusedImports] Avoid unused imports such as 'com.framgia.moviedb.data.source.MovieReposity'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenreActivity.java (hunk 14 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenreActivity.java]: WARNING: [UnusedImports] Avoid unused imports such as 'com.framgia.moviedb.data.source.remote.MovieRemoteDataSource'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenreActivity.java (hunk 15 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenreActivity.java]: WARNING: [UnusedImports] Avoid unused imports such as 'com.framgia.moviedb.data.source.remote.service.MovieServiceClient'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenreContract.java (hunk 17 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenreContract.java]: WARNING: [UnusedImports] Avoid unused imports such as 'com.framgia.moviedb.data.model.Movie'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenreContract.java (hunk 18 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenreContract.java]: WARNING: [UnusedImports] Avoid unused imports such as 'java.util.List'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenrePresenter.java (hunk 19 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenrePresenter.java]: WARNING: [UnusedImports] Avoid unused imports such as 'com.framgia.moviedb.BuildConfig'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenrePresenter.java (hunk 20 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenrePresenter.java]: WARNING: [UnusedImports] Avoid unused imports such as 'com.framgia.moviedb.data.model.Movie'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenrePresenter.java (hunk 21 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenrePresenter.java]: WARNING: [UnusedImports] Avoid unused imports such as 'com.framgia.moviedb.data.source.MovieReposity'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenrePresenter.java (hunk 22 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenrePresenter.java]: WARNING: [UnusedImports] Avoid unused imports such as 'io.reactivex.android.schedulers.AndroidSchedulers'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenrePresenter.java (hunk 23 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenrePresenter.java]: WARNING: [UnusedImports] Avoid unused imports such as 'io.reactivex.disposables.CompositeDisposable'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenrePresenter.java (hunk 24 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/moviesactivity/MoviesofgenrePresenter.java]: WARNING: [UnusedImports] Avoid unused imports such as 'io.reactivex.observers.DisposableObserver'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/moviedb/utils/binding/BindingUtils.java (hunk 28 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/utils/binding/BindingUtils.java]: ERROR: Unused import: android.widget.LinearLayout. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/moviedb/utils/binding/BindingUtils.java (hunk 29 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/utils/binding/BindingUtils.java]: ERROR: Unused import: com.framgia.moviedb.screen.movies.MoviesFragment. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/moviedb/utils/binding/BindingUtils.java (hunk 28 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/utils/binding/BindingUtils.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.widget.LinearLayout'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/moviedb/utils/binding/BindingUtils.java (hunk 29 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/utils/binding/BindingUtils.java]: WARNING: [UnusedImports] Avoid unused imports such as 'com.framgia.moviedb.screen.movies.MoviesFragment'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/moviedb/screen/movies/MoviesViewModel.java [hunk 30]
```diff
@@ -67,7 +69,7 @@ public void onStop() {
     @Override
     public void setPresenter(MoviesContract.Presenter presenter) {
         mPresenter = presenter;
-        mPresenter.getDataMovies(mCategory, mPage);
+        mPresenter.getDataMovies(mCategory, mPage,mId);
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/movies/MoviesViewModel.java]: ERROR: ',' is not followed by whitespace.

File: app/src/main/java/com/framgia/moviedb/screen/movies/MoviesViewModel.java [hunk 31]
```diff
@@ -113,7 +116,7 @@ public void setLoadingMore(boolean loadingMore) {
 
     public void loadMore() {
         mPage++;
-        mPresenter.getDataMovies(mCategory, mPage);
+        mPresenter.getDataMovies(mCategory, mPage,mId);
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/movies/MoviesViewModel.java]: ERROR: ',' is not followed by whitespace.

File: app/src/main/java/com/framgia/moviedb/utils/binding/BindingUtils.java (hunk 28 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/utils/binding/BindingUtils.java]: ERROR: Unused import: android.widget.LinearLayout. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/moviedb/utils/binding/BindingUtils.java (hunk 29 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/utils/binding/BindingUtils.java]: ERROR: Unused import: com.framgia.moviedb.screen.movies.MoviesFragment. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/moviedb/utils/binding/BindingUtils.java (hunk 28 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/utils/binding/BindingUtils.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.widget.LinearLayout'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/moviedb/utils/binding/BindingUtils.java (hunk 29 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/utils/binding/BindingUtils.java]: WARNING: [UnusedImports] Avoid unused imports such as 'com.framgia.moviedb.screen.movies.MoviesFragment'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/AndroidManifest.xml [hunk 32]
```diff
@@ -20,6 +20,10 @@
             android:name=".screen.detail.DetailActivity"
             android:screenOrientation="portrait"
             android:theme="@style/AppTheme"/>
+        <activity
+            android:name=".screen.moviesactivity.MoviesofgenreActivity"
```
Comment (DoanVanToan) [inline app/src/main/AndroidManifest.xml]: pls rename moviesactivity to movies
and Moviesofgenreactivity to MoviesActivity

File: app/src/main/java/com/framgia/moviedb/screen/detail/DetailViewModel.java [hunk 33]
```diff
@@ -119,6 +121,7 @@ public void setProductorAdapter(ProductorAdapter productorAdapter) {
 
     @Override
     public void onItemClickListener(Genre genre) {
-        // TODO: 28/09/2017
+        mContext.startActivity(MoviesofgenreActivity.getIntentMoviesOfGenre(mContext, genre)
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/detail/DetailViewModel.java]: pls use menthod .addFlag in MoviesofgenreActivity.getIntentMoviesOfGenre()

File: app/src/main/java/com/framgia/moviedb/screen/detail/DetailViewModel.java (hunk 1 above)
Comment (anhpham2403) [reply app/src/main/java/com/framgia/moviedb/screen/detail/DetailViewModel.java]: Em dùng import  này để chuyển activity

File: app/src/main/java/com/framgia/moviedb/screen/genres/GenresViewModel.java [hunk 34]
```diff
@@ -66,6 +68,7 @@ public void onDestroy() {
 
     @Override
     public void onItemClickListener(Genre genre) {
-        // TODO: 28/09/2017
+        mContext.startActivity(MoviesActivity.getIntentMoviesOfGenre(mContext, genre)
+                .addFlags(Intent.FLAG_ACTIVITY_NEW_TASK));
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/genres/GenresViewModel.java]: you definded this menthod in getIntentMoviesOfGenre

File: app/src/main/java/com/framgia/moviedb/screen/movies/MoviesFragment.java [hunk 35]
```diff
@@ -19,6 +19,7 @@
  */
 public class MoviesFragment extends BaseFragment {
     int category;
+    int idGenre;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/screen/movies/MoviesFragment.java]: mising access modifier

Comment (DoanVanToan) [reply app/src/main/java/com/framgia/moviedb/screen/movies/MoviesFragment.java]: the same for other variable

**Pull request merged**